### PR TITLE
Add noms commit --merge

### LINKED ISSUE #2535: Move noms-merge into `noms commit` tool
The new interface will be:

```
noms merge [--policy=l|r|a] [--out-ds-name=<name>] [--parent=<obj>] <db-spec> <left-obj> <right-obj>

  <db-spec>   : database in which named datasets live
  <left-obj>  : path to an object descending from <parent>
  <right-obj> : path to an object descending from <parent>

See Spelling Objects at https://github.com/attic-labs/noms/blob/master/doc/spelling.md for details on the <db-spec>, <left-obj>, <right-obj> and --parent arguments.

Flags:

--policy
    If unset, any merge conflict is fatal. Can be set to one of 'l', 'r', or 'a' which will resolve conflicts by choosing the Value from <left-obj>, <right-obj>, or asking the user respectively
--out-ds-name (= "")
    output dataset to write to - if empty, defaults to printing the hash of the merged object to stdout
--parent (= "")
    common ancestor of <left-obj> and <right-obj> (required unless left-obj and right-obj are Datasets)
--quiet  (= false)
    silence progress output
```

### PR DESCRIPTION
**Pull request by cmasone-attic**
Add optional merging functionality to noms commit.
noms commit --merge <database> <left-dataset-name> <right-dataset-name>

The command above will look in the given Database for the two named
Datasets and, if possible, merge their HeadValue()s and commit the
result back to <right-dataset-name>. All the existing options to
`noms commit` are supported, to allow users to add metadata to the
Commit object that winds up in the Database.

Users can optionally provide a third Dataset name to commit to instead.

Fixes #2535

### CODE REVIEW

**Comment by cmasone-attic** [comment]
PTAL

**Comment by aboodman** [comment]
I'd rather it was:

```
noms merge <db> <ds1> <ds2> [<out-ds>]
```

Sorry, it's ok to hate me. You'd not be able to set any commit metadata. You could use `poke` in some future world to do that if you wanted to.

**Comment by cmasone-attic** [comment]
Ok, PTAL :-)

File: cmd/noms/noms_merge.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	"github.com/attic-labs/noms/go/util/verbose"
+	flag "github.com/juju/gnuflag"
+)
+
+var (
+	resolver string
+
+	nomsMerge = &util.Command{
+		Run:       runMerge,
+		UsageLine: "merge [options] <database> <left-dataset-name> <right-dataset-name> [<output-dataset>]",
```
Comment (aboodman) [inline cmd/noms/noms_merge.go]: Let's just make the output dataset required.

File: cmd/noms/noms_merge.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		Long:      "See Spelling Objects at https://github.com/attic-labs/noms/blob/master/doc/spelling.md for details on the database argument.\nYu must provide a working database and at least two Dataset names. The values at the heads of these Datasets will be merged and put into a new Commit object. If you provide a third dataset name, its head will be set to this new merge commit. If not, the head of <right-dataset-name> will be set to this Commit instead.",
+		Flags:     setupMergeFlags,
+		Nargs:     1, // if absolute-path not present we read it from stdin
+	}
+	datasetRe = regexp.MustCompile("^" + datas.DatasetRe.String() + "$")
+)
+
+func setupMergeFlags() *flag.FlagSet {
+	commitFlagSet := flag.NewFlagSet("merge", flag.ExitOnError)
+	commitFlagSet.StringVar(&resolver, "policy", "n", "conflict resolution policy for merging. Defaults to 'n', which means no resolution strategy will be applied. Supported values are 'l' (left), 'r' (right) and 'a' (ask). 'ask' will bring up a simple command-line prompt allowing you to resolve conflicts by choosing between 'l' or 'r' on a case-by-case basis.")
```
Comment (aboodman) [inline cmd/noms/noms_merge.go]: Can can you change 'a' to 'p' (for 'prompt')

Comment (cmasone-attic) [inline cmd/noms/noms_merge.go]: Done

**Pull request merged**